src/core/common/debug.hpp

```diff
@@ -77,6 +77,7 @@
 #undef assert
 #endif
 
+#ifndef assert
```

Comment:
Is this necessary? Just a few lines above, there is a check and `#undef assert`.

Reply:
It is not necessary, I removed it.